lib/multiplexer/multiplexer.go

```diff
@@ -246,44 +256,146 @@ func (m *Mux) detectAndForward(conn net.Conn) {
 	listener.HandleConnection(m.context, connWrapper)
 }
 
-func detect(conn net.Conn, enableProxyProtocol bool) (*Conn, error) {
+type PROXYSigner interface {
+	SignPROXY(p jwt.PROXYSignParams) (string, error)
+}
+
+func getTCPAddr(a net.Addr) net.TCPAddr {
+	addr, ok := a.(*net.TCPAddr)
+	if ok { // Hot path
+		return *addr
+	}
+
+	parsedAddr := utils.FromAddr(a)
```

Comment:
A comment here explaining in what situation we expect `a` to be something other than a `net.TCPAddr` would be helpful :)

Reply:
We don't really expect some specific situation that causes it, that's why I added comment about hot path. But technically it might be something else, that implements the interface 🤷